atom/renderer/api/atom_api_spell_check_client.cc

```diff
@@ -74,14 +96,28 @@ void SpellCheckClient::RequestCheckingOfText(
     const blink::WebString& textToCheck,
     blink::WebTextCheckingCompletion* completionCallback) {
   base::string16 text(textToCheck.Utf16());
+  // Ignore invalid requests.
   if (text.empty() || !HasWordCharacters(text, 0)) {
     completionCallback->DidCancelCheckingText();
     return;
   }
 
-  std::vector<blink::WebTextCheckingResult> results;
-  SpellCheckText(text, false, &results);
-  completionCallback->DidFinishCheckingText(results);
+  // Clean up the previous request before starting a new request.
+  if (pending_request_param_.get()) {
+    pending_request_param_->completion()->DidCancelCheckingText();
+  }
+
+  pending_request_param_.reset(new SpellcheckRequest(text, completionCallback));
+
+  SpellcheckRequest* request = pending_request_param_.release();
+
+  if (!request) {
```

Comment:
How is `!request` ever true here?

Reply:
for same reason of https://cs.chromium.org/chromium/src/components/spellcheck/renderer/spellcheck.cc?sq=package:chromium&dr=C&l=414-423, I preserved most of safety check as-is. (In Chromium, it resets value, then call `postDealyedTask` immediately by releasing value. but in `postDealyedTask` it checks `!request` there. I have merged `postDealyedTask` into single fn but preserved check there)

Reply:
Thank you for the link, that makes things a little clearer.

I still am not sure the code is correct in this PR though...?

In the link, the answer to "How is `!request` ever true" is the `if (InitializeNeeded()) return;`  branch that could leave `pending_request_param_` in a nonempty state. That branch doesn't exist in this PR, so it still reads to me that line 106 will never be true...?

Reply:
Yes, I'm not saying this PR's change's legit, just background how I did and most likely I missed to update few things correctly. When I moved over request guard check, I picked it up just in cases if I might misses places other than init check could make `request` invalid - and I may wrong there.